glib/src/object.rs

```diff
@@ -3267,6 +3294,87 @@ wrapper! {
     }
 }
 
+struct WeakRefNotifyData {
+    callback: Box<dyn Fn()>,
+    should_drop: AtomicBool,
+}
+
+impl WeakRefNotifyData {
+    pub fn new<F: Fn() + 'static>(f: F) -> ManuallyDrop<Pin<Box<Self>>> {
+        ManuallyDrop::new(Box::pin(WeakRefNotifyData {
+            callback: Box::new(f),
+            should_drop: AtomicBool::new(false),
+        }))
+    }
+
+    // rustdoc-stripper-ignore-next
+    /// Checks whether the data should be dropped and marks it to be dropped
+    pub fn should_drop(&self) -> bool {
+        self.should_drop.swap(true, Ordering::SeqCst)
+    }
+}
+
+// rustdoc-stripper-ignore-next
+/// A handle to disconnect a weak ref notify closure.
+pub struct WeakRefNotify {
+    object: WeakRef<Object>,
```

Comment:
`impl Deref<Target=WeakRef>` would make sense here but then also make make it an `WeakRefNotify<T>` :)

Reply:
how would `WeakRef::set()` work? if the user changes the object inside the `WeakRef` it would break `WeakRefNotify::disconnect()`

Reply:
Indeed, you can only provide an `upgrade()` function but not `Deref`